feat: Import Relay docs (#2197)
getsentry/sentry-docs: src/docs/product/relay/options.mdx

---
title: Configuration Options
sidebar_order: 2
---

The base configuration for Relay lives in the file `.relay/config.yml`. To change this location, pass the `--config` option to any Relay command:

```sh
❯ ./relay run --config /path/to/folder
```

All configuration keys are `snake_case`.

## Relay

> General relay settings for Relay's operation.

Reviewer: ```suggestion
General settings for Relay's operation.
```

upstream before Relay starts buffering requests. Queries are all requests made
to the upstream for obtaining information and explicitly exclude event
submission.

The concurrency of queries is additionally constrained by `max_concurrent_requests`.

### `limits.max_event_size`

_String, default: `1MiB`_

The maximum payload size for events.

### `limits.max_attachment_size`

_String, default: `50MiB`_

The maximum size for each attachment.

### `limits.max_attachments_size`

_String, default: `50MiB`_

The maximum combined size for all attachments in an envelope or request.

### `limits.max_envelope_size`

_String, default: `50MiB`_

The maximum payload size for an entire envelopes. Individual limits still apply.

### `limits.max_session_count`

_Integer, default: `100`_

The maximum number of session items per envelope.

### `limits.max_api_payload_size`

_String, default: `20MiB`_

The maximum payload size for general API requests.

### `limits.max_api_file_upload_size`

_String, default: `40MiB`_

The maximum payload size for file uploads and chunks.

### `limits.max_api_chunk_upload_size`

_String, default: `100MiB`_

> The maximum payload size for chunks

Reviewer: ```suggestion
The maximum payload size for chunks.
```

### `limits.max_thread_count`

_Integer, default: number of cpus_

> The maximum number of threads to spawn for CPU and web worker, each.

Reviewer: ```suggestion
The maximum number of threads to spawn for each CPU and web worker.
```

- `pretty`: Human readable format with colors
- `simplified`: Simplified human readable log output
- `json`: JSON records, suitable for logging software

### `logging.enable_backtraces`

_boolean, default: `true`_

Writes back traces for all internal errors to the log stream and includes them in Sentry errors, if enabled.

## Statsd Metrics

### `metrics.statsd`

_String, optional_

If set to a host/port string then metrics will be reported to this statsd
instance.

### `metrics.prefix`

_String, default: `sentry.relay`_

The prefix that should be added to all metrics.

### `metrics.default_tags`

_Map of strings to strings, default empty_

A set of default tags that should be attached to all outgoing statsd metrics.

### `metrics.hostname_tag`

_String, optional_

If set, reports the current hostname under the given tag name for all metrics.

## Internal Error Reporting

Configures error reporting for errors happening within Relay. Disabled by
default.

### `sentry.enabled`

_boolean, default: `false`_

Whether to report internal errors to a separate DSN. `false` means no internal
errors are sent, but still logged.

### `sentry.dsn`

_String, optional_

> DSN to report internal Relay failures to.

Reviewer: ```suggestion
DSN to which to report internal Relay failures.
```

> It is not a good idea to set this to a value that will make the Relay send

Reviewer: ```suggestion
We recommend setting this to a value that will not send Relay errors to itself. Ideally, this value should send errors to Sentry directly, not another Relay.
```

> errors to itself. Ideally this should just send errors to Sentry directly, not

Reviewer: remove (incorporated into previous edit)

> another Relay.

Reviewer: remove (incorporated into previous edit)
